src/basic_recipes/barplot.jl

```diff
@@ -160,7 +161,35 @@ function stack_grouped_from_to(i_stack, y, grp)
     (from = from, to = to)
 end
 
-function text_attributes(values, in_y_direction, flip_labels_at, color_over_background, color_over_bar, label_offset)
+function calculate_bar_label_align(label_align, label_rotation::Real, in_y_direction::Bool, flip::Bool)
+    is_align_type(::Any) = false
+    is_align_type(::Vec2f) = true
+    is_align_type(::Point2f) = true
+    is_align_type(::NTuple{2, <:Real}) = true
+    is_align_type(::Tuple{<:Real, <:Real}) = true
+    is_align_type(::NTuple{2, Symbol}) = true
+    make_align(a) = Vec2f(a)
+    make_align(a::NTuple{2, Symbol}) = to_align(a)
+    if label_align == automatic
+        if flip
+            label_rotation += π
+        end
+        if !in_y_direction
+            label_rotation += π/2
+        end
+        s, c = sincos(label_rotation)
+        scale = 1 / max(abs(s), abs(c))
+        align = Vec2f(0.5 - 0.5scale * s, 0.5 - 0.5scale * c)
+        return align
+    elseif is_align_type(label_align)
+        return make_align(label_align)
+    else
+        error("`label_align` needs to be of type NTuple{2, <:Real}, not of type $(typeof(label_align))")
```

Comment:
You can simplify this to something like
```julia
make_align(a::VecTypes{2, <:Real)) = Vec2f(a)
make_align(a::NTuple{2, Symbol}) = to_align(a)
make_align(a) = error(...)
...
else
    return make_align(a)
end
```
(i.e. drop the type check functions)

Reply:
ah good one!

Reply:
I have pushed a new commit to address your comment